app.py

```diff
@@ -27,6 +27,10 @@ def init_app(self):
         self._tk_multi_workfiles = self.import_module("tk_multi_workfiles")
         self.__is_pyside_unstable = None
 
+        if not self.engine.has_ui:
+            self.log_debug("The engine reports that there is no UI. Workfiles2 will not continue initializing.")
```

Comment:
Good call on the log line, however I think we should avoid this method call specifically, as it is [deprecated](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=log_debug#sgtk.platform.Application.log_debug).

Should it be `self.logger.debug(...` instead?